base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
```

Comment:
But then what's are the boundaries of this file? Any function uses an "algorithm". Maybe we'd need a collections.jl file with all generic functions which apply to any collection. Anyway, better use an existing file for this PR, and open another one to regroup functions in a subsequent PR.